### ISSUE TITLE
scancodes in command palette

### ISSUE THREAD

**Issue by stanciuadrian** [microsoft/terminal#12123]
### Windows Terminal version

12.3472.0

### Windows build number

10.0.22000.376

### Steps to reproduce

Press ctrl+shift+p
Type "qua"
Notice the "Summon Quake window" shortcut

### Expected Behavior

The shortcut text should be "Win + `". I don't know the scancodes by heart.

### Actual Behavior

The shortcut text is: `win + sc(41)`.

**Comment by zadjii-msft** [comment]
Huh. This is a good point. We switched that to being bound to a scancode instead of the character, so that it would work on non-en-US layouts the same, but the `sc()` notation is a little hard to parse.


I wonder if there's a way for us to store the `sc` internally, but still display the character in the SUI&cmdpal...

**Comment by mauve** [comment]
you can convert the scancode to a keyname using the win32 API `GetKeyNameText()` which works but has a somewhat annoying API, here is an example which converts the example in the issue report `sc(41)` to `` ` ``:

```cpp
std::string scan_code_to_string(unsigned int scan_code)
{
    std::vector<char> buffer;
    buffer.resize(64);
    if (!GetKeyNameTextA(
            static_cast<unsigned long>(scan_code) << 16,
            buffer.data(),
            static_cast<int>(buffer.size())))
    {
        throw std::runtime_error{"cannot get key name"};
    }

    return {buffer.data()};
}
```

When run in a very small hack program I get the following output:

![image](https://user-images.githubusercontent.com/795800/151704624-4f64a3cf-1ee1-405d-b2d9-df94db2c8418.png)

**Comment by mauve** [comment]
Maybe I can fix this, but I haven't touched the code in quite some while, is this the line where the conversion would need to happen https://github.com/microsoft/terminal/blob/main/src/cascadia/TerminalApp/ActionPaletteItem.cpp#L41?

Maybe something like this? @DHowett  

```cpp
-        item->KeyChordText(senderCommand.KeyChordText());
+        item->KeyChordText(find_and_replace_scancodes(senderCommand.KeyChordText()));
```

If you point me in the correct direction I can whip up a patch next weekend.

**Comment by DHowett** [comment]
this looks great to me! if it ends up being the wrong place, we can iterate on it together in the PR :)

**Comment by mauve** [comment]
@DHowett I created a PR, it works on my machine :) not super-happy with the code, but it works.
